### ISSUE TITLE
Type hinting (PHPDoc and method signatures)

### ISSUE THREAD

**Issue by jmfeurprier** [mapado/rest-client-sdk#41]
Type hinting is a bit off on most classes of this project, which makes it difficult to use in recent IDEs.

```
class EntityRepository
{
// ...
    /**
     * @object REST Client
     */
    protected $restClient;

    /**
     * @var SDK object
     */
    protected $sdk;
// ...
    /**
     * EntityRepository constructor
     *
     * @param object $sdkClient - the client to connect to the datasource with
     * @param object $restClient - client to process the http requests
     * @param string $entityName The entiy to work with
     */
    public function __construct($sdkClient, $restClient, $entityName)
    {
        $this->sdk = $sdkClient;
        $this->restClient = $restClient;
        $this->entityName = $entityName;
    }
// ...
}
```

I could take care of this by adding/fixing PHPDoc blocks and methods signatures as such (for example):

```
class EntityRepository
{
// ...
    /**
     * @var RestClient
     */
    protected $restClient;

    /**
     * @var SdkClient
     */
    protected $sdk;
// ...
    /**
     * EntityRepository constructor
     *
     * @param SdkClient  $sdkClient  The client to connect to the datasource with
     * @param RestClient $restClient The client to process the http requests
     * @param string     $entityName The entity to work with
     */
    public function __construct(SdkClient $sdkClient, RestClient $restClient, $entityName)
    {
        $this->sdk = $sdkClient;
        $this->restClient = $restClient;
        $this->entityName = $entityName;
    }
// ...
}
```

Just let me know if it's ok with you :)

**Comment by jdeniau** [comment]
Great ! Thank you ! Waiting for your PR

**Comment by jmfeurprier** [comment]
Great ! Just a question before I proceed : do you want to keep PHPDoc @access annotations (which are from PHP4, if I remember well) ?

**Comment by jdeniau** [comment]
If possible yes, as it is declared in other phpdocs

**Comment by jmfeurprier** [comment]
Added pull request #42

**Comment by jdeniau** [comment]
Fixed by #42
